gui/src/main/java/com/devonfw/ide/gui/MainController.java

```diff
@@ -0,0 +1,132 @@
+import com.devonfw.tools.ide.log.IdeLogListenerBuffer;
+import com.devonfw.tools.ide.log.IdeSubLoggerOut;
+
+/**
+ *
+ */
+public class MainController {
+
+  @FXML
+  private ComboBox<String> selectedProject;
+  @FXML
+  private ComboBox<String> selectedWorkspace;
+  @FXML
+  private Button androidStudioOpen;
+  @FXML
+  private Button eclipseOpen;
+  @FXML
+  private Button intellijOpen;
+  @FXML
+  private Button vsCodeOpen;
+
+  private final String directoryPath = "C:\\projects\\";
+  private String projectValue;
+  private String workspaceValue;
+
+
+  @FXML
+  private void initialize() {
+
+    setProjectsComboBox(selectedProject);
+  }
+
+  @FXML
+  private void openAndroidStudio() {
+
+    openIDE("android-studio");
+  }
+
+  @FXML
+  private void openEclipse() {
+
+    openIDE("eclipse");
+  }
+
+  @FXML
+  private void openIntellij() {
+
+    openIDE("intellij");
+  }
+
+  @FXML
+  private void openVsCode() {
+
+    openIDE("vscode");
+  }
+
+
+  private void setProjectsComboBox(ComboBox<String> projects) {
+
+    projects.getItems().clear();
+    File directory = new File(directoryPath);
+    if (directory.exists() && directory.isDirectory()) {
+
+      File[] subDirectories = directory.listFiles(File::isDirectory);
+      if (subDirectories != null) {
+
+        for (File subDirectory : subDirectories) {
+
+          String name = subDirectory.getName();
+          if (!name.startsWith("_")) {
+
+            projects.getItems().add(name);
+          }
+        }
+      }
+    }
+
+    projects.setOnAction(actionEvent -> {
+
+      projectValue = projects.getValue() + "\\workspaces";
+      setWorkspaceValue(selectedWorkspace);
+      selectedWorkspace.setDisable(false);
+      androidStudioOpen.setDisable(false);
+      eclipseOpen.setDisable(false);
+      intellijOpen.setDisable(false);
+      vsCodeOpen.setDisable(false);
+    });
+  }
+
+  private void setWorkspaceValue(ComboBox<String> workspace) {
+
+    workspace.getItems().clear();
+    File directory = new File(directoryPath + projectValue);
+    if (directory.exists() && directory.isDirectory()) {
+
+      File[] subDirectories = directory.listFiles(File::isDirectory);
+      if (subDirectories != null) {
+
+        for (File subDirectory : subDirectories) {
+
+          String name = subDirectory.getName();
+          workspace.getItems().add(name);
+          workspace.setValue("main");
+        }
+      }
+    }
+
+    workspace.setOnAction(actionEvent -> {
+
+      workspaceValue = workspace.getValue();
+    });
+  }
+
+  private void openIDE(String inIde) {
+
+    final IdeLogListenerBuffer buffer = new IdeLogListenerBuffer();
+    IdeLogLevel logLevel = IdeLogLevel.INFO;
+    IdeStartContextImpl startContext = new IdeStartContextImpl(logLevel, level -> new IdeSubLoggerOut(level, null, true, logLevel, buffer));
+    IdeGuiContext context = new IdeGuiContext(startContext, Path.of(directoryPath + projectValue + workspaceValue));
```

Comment:
This code looks a little misplaced here.
IMHO we will have many different use-cases over time in the dashboard.
Please extract a private method to get the context as otherwise sooner or later this will be copy&pasted for other use-cases. Instead we want to keep control over the context creation and may avoid recreating the context for every use-case invocation later.
Also buffering the logging does not seem to make sense here (`IdeLogListenerBuffer`).